app/models/order_details_updater.rb

```diff
@@ -84,4 +90,12 @@ def find_value(item_id)
     Item.where(id: params[:order][:order_details][:item_id]).find_each { |item| @cache[item.id] = item }
     @cache[item_id].value
   end
+
+  def deduct_from_inventory
+    order_details.each do |order_detail|
+      item = order_detail.item
+      item.current_quantity -= order_detail.quantity
+      item.commit
```

Comment:
What does `commit` do? Did you intend `save` instead? Specifically, I think this should be `item.save!` which will throw an error if there is a failure. This is especially important because the error will cause the transaction to fail and all changes will be rolled back.